web/models/habit.ex

```diff
@@ -79,4 +79,24 @@ defmodule Habits.Habit do
     |> assoc(:check_ins)
     |> Repo.count
   end
+
+  def check_in_data(habit) do
+    query = """
+    SELECT
+      COUNT(check_ins.*)
+    FROM generate_series((
+      SELECT MIN(date_trunc('week', check_ins.date))
+      FROM check_ins
+      WHERE check_ins.habit_id = $1
+    ), NOW(), '1 week'::interval) week
+    LEFT OUTER JOIN check_ins
+      ON date_trunc('week', check_ins.date) = week
+      AND check_ins.habit_id = $1
+    GROUP BY week
+    ORDER BY week
+    ;
+    """
+    %Postgrex.Result{rows: rows} = Ecto.Adapters.SQL.query!(Repo, query, [habit.id])
+    Enum.map(rows, &List.first/1)
+  end
```

Comment:
This is hideous, but resisted my initial efforts at writing it in Ecto.

- Curiously, I got `undefined function fragment` when trying to use [`fragment/1`](https://hexdocs.pm/ecto/Ecto.Query.API.html#fragment/1) for the `generate_series`, even though `Ecto.Query` is `import`ed in models like this. Maybe I need to update `web.ex` for newer versions of Phoenix/Ecto.
- Even when referencing it directly as `Ecto.Query.API.fragment/1`, I couldn't get the Ecto syntax for joins (and nested fragments for `date_trunc` right

I also considered extracting a database view like I did for streaks, but this query depends on an argument (the habit) used in two places, and database views can't take arguments. Often, you can build the query in such a way that the argument can be replaced by a further `where` condition, but not in this case where it's needed in both a subquery and `ON` specification. So I think I'll let this stay ugly for a bit, as it works beautifully.